feature/onboarding/src/main/java/com/napzak/market/onboarding/nickname/NicknameScreen.kt

```diff
@@ -0,0 +1,230 @@
+package com.napzak.market.onboarding.nickname
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.PaddingValues
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material3.Icon
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.vector.ImageVector
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.res.vectorResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.hilt.navigation.compose.hiltViewModel
+import com.napzak.market.designsystem.R.drawable.ic_arrow_left
+import com.napzak.market.designsystem.R.drawable.ic_second_step_indicator
+import com.napzak.market.designsystem.component.button.NapzakButton
+import com.napzak.market.designsystem.component.textfield.NapzakDefaultTextField
+import com.napzak.market.designsystem.theme.NapzakMarketTheme
+import com.napzak.market.feature.onboarding.R.string.onboarding_next
+import com.napzak.market.feature.onboarding.R.string.onboarding_nickname_edit_hint
+import com.napzak.market.feature.onboarding.R.string.onboarding_nickname_sub_title
+import com.napzak.market.feature.onboarding.R.string.onboarding_nickname_success
+import com.napzak.market.feature.onboarding.R.string.onboarding_nickname_title
+import com.napzak.market.feature.onboarding.R.string.onboarding_nickname_validation_button
+import com.napzak.market.onboarding.nickname.model.NicknameUiState
+import com.napzak.market.onboarding.nickname.model.NicknameValidationResult
+import com.napzak.market.util.android.noRippleClickable
+
+@Composable
+internal fun NicknameRoute(
+    onBackClick: () -> Unit,
+    onNextClick: () -> Unit,
+) {
+    val viewModel: NicknameViewModel = hiltViewModel()
+
+    NicknameScreen(
+        uiState = viewModel.uiState.collectAsState().value,
```

Comment:
p1: `.collectAsStateWithLifecycle`로 NicknameRoute 함수의 수명주기에 맞게 안정적으로 상태를 불러올 수 있어요!
```suggestion
        uiState = viewModel.uiState.collectAsStateWithLifecycle().value,
```

Reply:
수정하겠습니담!!